web/tools/build.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+#
+# Compile KeymanWeb's dev & test tool modules
+#
+set -eu
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
+. "$(dirname "$THIS_SCRIPT")/../../resources/build/build-utils.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
+# This script runs from its own folder
+cd "$(dirname "$THIS_SCRIPT")"
+
+################################ Main script ################################
+
+# Note:  common/web/recorder is only needed for :recorder, but the builder can't do target-only dependencies yet.
+
+builder_describe "Builds the Keyman Engine for Web's development & unit-testing tools" \
+  "@../../common/web/keyman-version" \
+  "@../../common/web/keyboard-processor" \
+  "@../../common/web/recorder" \
+  "clean" \
+  "configure" \
+  "build" \
+  ":device     Builds the device-detect submodule" \
+  ":recorder   Builds the KMW recorder submodule for development of unit-test resources" \
+  ":wrappers   Builds the submodule for isolated testing of the outputTarget wrappers for various elements"
+
+builder_describe_outputs \
+  configure:device   /node_modules \
+  configure:recorder /node_modules \
+  configure:wrapper  /node_modules \
+  build:device       device-detect/build/index.js \
+  build:recorder     recorder/build/index.js \
+  build:wrappers     element-wrappers/build/index.js
```

Comment:
Conceptually, we could split this into three build.sh scripts, one for each target, and then have a parent script which just calls the three? Not sure if we need that level of structure at this point for tooling folders like this, but something to think about.

Reply:
I suppose we _could_; I just didn't see much of a point given how simple and straight-forward each one's build is.

Also note that I actually eliminated the previous one for `build:recorder` since it was legacy-style.

Reply:
It's more about modularization and consistency than efficiency. If each module has its own build script, there's less for devs to remember. But I don't think we need to necessarily do this, just a consideration.